v2/yarpcgrpc/integration_test.go

```diff
@@ -21,521 +21,391 @@
+		},
+		{
+			name:                 "larger than server send",
+			value:                strings.Repeat("a", 10),
+			ServerMaxSendMsgSize: 1,
+			errCode:              yarpcerror.CodeResourceExhausted,
+		},
+		{
+			name:                 "larger than client recieve",
+			value:                strings.Repeat("a", 10),
+			ClientMaxRecvMsgSize: 1,
+			errCode:              yarpcerror.CodeResourceExhausted,
 		},
 	}
 
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			scenario := createTLSScenario(t, test.clientValidity, test.serverValidity)
-
-			serverCreds := credentials.NewTLS(&tls.Config{
-				GetCertificate: func(_ *tls.ClientHelloInfo) (*tls.Certificate, error) {
-					return &tls.Certificate{
-						Certificate: [][]byte{scenario.ServerCert.Raw},
-						Leaf:        scenario.ServerCert,
-						PrivateKey:  scenario.ServerKey,
-					}, nil
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			te := testEnvOptions{
+				Procedures: yarpcjson.Procedure("test-procedure", testEchoHandler),
+				Inbound: &Inbound{
+					ServerMaxRecvMsgSize: tt.ServerMaxRecvMsgSize,
+					ServerMaxSendMsgSize: tt.ServerMaxSendMsgSize,
 				},
-				ClientAuth: tls.RequireAndVerifyClientCert,
-				ClientCAs:  scenario.CAs,
-			})
-
-			clientCreds := credentials.NewTLS(&tls.Config{
-				GetClientCertificate: func(_ *tls.CertificateRequestInfo) (*tls.Certificate, error) {
-					return &tls.Certificate{
-						Certificate: [][]byte{scenario.ClientCert.Raw},
-						Leaf:        scenario.ClientCert,
-						PrivateKey:  scenario.ClientKey,
-					}, nil
+				Dialer: &Dialer{
+					ClientMaxRecvMsgSize: tt.ClientMaxRecvMsgSize,
+					ClientMaxSendMsgSize: tt.ClientMaxSendMsgSize,
 				},
-				RootCAs: scenario.CAs,
-			})
-
-			te := testEnvOptions{
-				InboundOptions: []InboundOption{InboundCredentials(serverCreds)},
-				DialOptions:    []DialOption{DialerCredentials(clientCreds)},
 			}
-			te.do(t, func(t *testing.T, e *testEnv) {
-				err := e.SetValueYARPC(context.Background(), "foo", "bar")
-				expectErrorContains(t, err, test.expectedErrContains)
 
-				err = e.SetValueGRPC(context.Background(), "foo", "bar")
-				expectErrorContains(t, err, test.expectedErrContains)
+			doWithTestEnv(t, te, func(t *testing.T, testEnv *testEnv) {
+				client := yarpcjson.New(testEnv.Client)
+				ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+				defer cancel()
+
+				var res testEchoResponse
+				err := client.Call(ctx, "test-procedure", &testEchoRequest{Message: tt.value}, &res)
+				require.Equal(t, tt.errCode.String(), yarpcerror.FromError(err).Code().String())
 			})
 		})
 	}
 }
 
-func expectErrorContains(t *testing.T, err error, contains string) {
-	if contains == "" {
-		assert.NoError(t, err)
-	} else {
-		require.Error(t, err)
-		assert.Contains(t, err.Error(), contains)
+func TestJSONRoundTrip(t *testing.T) {
+	t.Parallel()
+
+	const procedure = "test-procedure"
```

Comment:
nit: Could we use `"test-procedure"` in the `yarpcjson.Procedure` call explicitly, and add this to the `basic` and `echo err` test cases? That way we can drop the lingering constant. 

In general I appreciate constants when they make things more readable, but this looks a little funky to me.